edx/edx-enterprise
# ENT-3549 Multiple IDP's can be attached with enterprise customer

We need to support multiple IDP's with one Enterprise Customer. Previously we had a one-to-one relationship of enterprise_customer and identity_provider. In this PR we have changed this relationship and added the foreign key of enterprise_customer in enterprise_identity_provider. 
Also, to allow nested inline support we have added extra = 0 in `enterprise/admin/__init__.py` file.

P.S: Tests will be fixed once the complete work for this epic is done.

How to test:
1- Please login in django admin.
2- Please go to the [link](http://localhost:18000/admin/enterprise/enterprisecustomer/) and open any enterprisecustomer. 
3- Please scroll down to the section `ENTERPRISE CUSTOMER IDENTITY PROVIDERS` and here you should see the `Add another Enterprise customer identity provider` button and should be able to attach multiple identity providers with enterprise customers.

<img width="1330" alt="Screenshot 2020-11-17 at 1 57 24 PM" src="https://user-images.githubusercontent.com/11922730/99382191-f7854a80-28ed-11eb-80e8-78772a55864d.png">

Add another Enterprise customer identity provider` button is added in this PR and the following image shows how it looks before this PR.

<img width="1326" alt="Screenshot 2020-11-17 at 1 53 58 PM" src="https://user-images.githubusercontent.com/11922730/99382298-2996ac80-28ee-11eb-81cf-8bf6256498dc.png">


**If you try to link one IDP to two Enterprise Customers then it will give an error like the following image.**

![Screenshot 2020-11-17 at 3 30 08 PM](https://user-images.githubusercontent.com/11922730/99387551-0112b080-28f6-11eb-80db-9b1d34be5a41.png)


**Merge checklist:**
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

**Post merge:**
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.
